src/Data/List/Membership/Setoid.agda

```diff
@@ -48,9 +54,10 @@ mapWith∈ (x ∷ xs) f = f (here refl) ∷ mapWith∈ xs (f ∘ there)
 
 module _ {p} {P : Pred A p} where
 
-  find : ∀ {xs} → Any P xs → ∃ λ x → x ∈ xs × P x
-  find (here px)   = (_ , here refl , px)
+  find : ∀ {xs} → Any P xs → ∃∈ P xs
+  find (here px)   = _ , here refl , px
   find (there pxs) = Product.map id (Product.map there id) (find pxs)
+  -- better? let x , x∈xs , px = find pxs in x , there x∈xs , px
```

Comment:
Done. I'd left in the comment in the hope it might get picked up. I think that with the 'better' definition, further simplifications may  be possible, but haven't had the headspace to pursue that.